tests/local.rs

```diff
@@ -0,0 +1,351 @@
+#[macro_use]
+extern crate serial_test_derive;
+extern crate fantoccini;
+extern crate futures_util;
+
+use fantoccini::{error, Client};
+use std::net::{Ipv4Addr, TcpListener, SocketAddr};
+use std::path::PathBuf;
+use warp::Filter;
+use futures_core::Future;
+use fantoccini::error::CmdError;
+use warp::test::request;
+
+
+async fn select_client_type(s: &str) -> Result<Client, error::NewSessionError> {
```

Comment:
Ah, sorry, what I meant by factor this out was to _also_ share it with the same code in `tests/common.rs` here:
https://github.com/jonhoo/fantoccini/blob/28087fb4e3ef5457c2400194c157a2818a46d435/tests/common.rs#L13

It seems unfortunate for us to have to do this complex matching + construction  in the same way in two different files.

Reply:
I've moved the remote testing functions to another file and now moved all setup fns and macros to common.